src/main/java/com/faforever/client/replay/ReplayCardController.java

```diff
@@ -136,13 +149,31 @@ protected void onInitialize() {
                     .map(reviewsSummary -> reviewsSummary.score() / reviewsSummary.numReviews())
             .when(showing));
 
+
+    isReplayWatched.bind(Bindings.createBooleanBinding(() -> {
+      Integer id = entity.map(Replay::id).getValue();
+      ObservableSet<Integer> replayHistory = this.replayHistory.getWatchedReplays();
+      if( id != null && replayHistory != null ){
+        return replayHistory.contains(id);
+      } else {
+        return false;
+      }
+    }, entity, replayHistory.getWatchedReplays()));
```

Comment:
The null checks can be simplified here a little bit as well as the dependencies
```suggestion
    isReplayWatched.bind(replayHistory.watchedReplaysProperty().flatMap(watchedReplays -> entity.map(Replay::id).map(watchedReplays::contains)).orElse(false));
```